### ISSUE TITLE
Cross-Origin-Embedder-Policy

### ISSUE THREAD

**Issue by Malvoz** [helmetjs/helmet#198]
There's a very early draft of [`Cross-Origin-Embedder-Policy` (COEP)](https://mikewest.github.io/corpp/).
However, I think it's worth tracking at this point. 

[This document](https://docs.google.com/document/d/1zDlfvfTJ_9e8Jdc8ehuV4zMEu9ySMCiTGMS9y0GU92k/mobilebasic#) also explains the relation to [CORP](https://github.com/helmetjs/helmet/issues/176), and [COOP](https://github.com/helmetjs/helmet/issues/177).

**Comment by EvanHahn** [comment]
Agreed—we'll keep an eye on this. I'll check on the spec every so often, but please let me know if you spot something before I check.

**Comment by Malvoz** [comment]
Chrome status: https://chromestatus.com/feature/5642721685405696

**Comment by EvanHahn** [comment]
Thanks! It looks like this is in development but not supported yet, so I'll wait to add this.

**Comment by calebmer** [comment]
This along with `Cross-Origin-Opener-Policy` look like they’ll be required to enable `SharedArrayBuffer` around May 2021 https://developer.chrome.com/blog/enabling-shared-array-buffer/

**Comment by EvanHahn** [comment]
I've added support for `Cross-Origin-Embedder-Policy` (among other `Cross-Origin-` headers) in v4.5.0-rc.1.

You can install with `npm install helmet@4.5.0-rc.1`, see [the docs for 4.5.x on GitHub](https://github.com/helmetjs/helmet/tree/v4.5.x), and the pull request at #296.

Please try it and give me feedback! I will plan to release on April 17/18 unless something disrupts that, such as significant errors in my implementation.

**Comment by EvanHahn** [comment]
This was released in `helmet@4.5.0`. Check out the documentation to see how to use the header.

**Comment by sricharankrishnan** [comment]
Hello Team Helmet!
Thank you so much for this awesome piece of tech you guys have built. I've just been recently exposed to various HTTP headers that can help improve application security. I came across this article written by Scott Helme @ https://scotthelme.co.uk/coop-and-coep/

My question is: can we get a Cross-Origin-Embedder-Policy-Report-Only and Cross-Origin-Opener-Policy-Report-Only mode for helmet. Even for the current version. I think it would be a great add on for us to understand these features.

What are your thoughts on this please?

Cheers and grateful once again for your support and kindness.

**Comment by sricharankrishnan** [comment]
Hi Evan!
Done as requested. The title of the request is
**COOP and COEP - Report Only Mode Feature Request #336**
Cheers.

**Comment by Sahillather002** [comment]
Error: Cross-Origin-Embedder-Policy does not support the "cross-origin" policy
Can anyone help me why this error!

**Comment by EvanHahn** [comment]
> Error: Cross-Origin-Embedder-Policy does not support the "cross-origin" policy
> 
> Can anyone help me why this error!

@Sahillather002 I think this is happening because Cross-Origin-Embedder-Policy doesn't support the "cross-origin" directive. You can see more about this header on [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Embedder-Policy) docs.

If you're still running into problems, please open a new issue.

**Comment by Sahillather002** [comment]
throw new Error(`Cross-Origin-Embedder-Policy does not support the ${JSON.stringify(policy)} policy`)
                      ^

Error: Cross-Origin-Embedder-Policy does not support the "cross-origin" policy
    at getHeaderValueFromOptions$6 (file:///F:/project/fullapp/server/node_modules/helmet/index.mjs:120:9)
    at Function.crossOriginEmbedderPolicy (file:///F:/project/fullapp/server/node_modules/helmet/index.mjs:124:22)
    at file:///F:/project/fullapp/server/index.js:19:16
    at ModuleJob.run (node:internal/modules/esm/module_job:198:25)      
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:385:24) 
    at async loadESM (node:internal/process/esm_loader:88:5)
    at async handleMainPromise (node:internal/modules/run_main:61:12)   
[nodemon] app crashed - waiting for file changes before starting...

This is the exact error is couldn't get over it!
